File: add_ll.c
#include <stdio.h>
#include <stdlib.h>
struct Node {
    int data;
    struct Node* next;
};
struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}
void printList(struct Node* node) {
    while (node != NULL) {
        printf("%d ", node->data);
        node = node->next;
    }
    printf("\n");
}
struct Node* addTwoLists(struct Node* first, struct Node* second) {
    struct Node* res = NULL;
    struct Node** lastPtrRef = &res;
    int carry = 0, sum;
    while (first != NULL || second != NULL) {
        int a = (first) ? first->data : 0;
        int b = (second) ? second->data : 0;
        sum = carry + a + b;
        carry = sum / 10;
        sum = sum % 10;
        *lastPtrRef = newNode(sum);
        lastPtrRef = &((*lastPtrRef)->next);
        if (first) first = first->next;
        if (second) second = second->next;
    }
    if (carry > 0) {
        *lastPtrRef = newNode(carry);
    }
    return res;
}
int main() {
    struct Node* first = newNode(2);
    first->next = newNode(4);
    first->next->next = newNode(3);
    struct Node* second = newNode(5);
    second->next = newNode(6);
    second->next->next = newNode(4);
    struct Node* result = addTwoLists(first, second);
    printf("Sum List: ");
    printList(result);
    return 0;
}

File: add_ll.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def addTwoLists(first, second):
    res = None
    last = None
    carry = 0
    while first or second:
        a = first.data if first else 0
        b = second.data if second else 0
        s = carry + a + b
        carry = s // 10
        s = s % 10
        newNode = Node(s)
        if res is None:
            res = newNode
            last = res
        else:
            last.next = newNode
            last = newNode
        if first:
            first = first.next
        if second:
            second = second.next
    if carry > 0:
        last.next = Node(carry)
    return res
def printList(node):
    while node:
        print(node.data, end=" ")
        node = node.next
    print()
if __name__ == "__main__":
    first = Node(2)
    first.next = Node(4)
    first.next.next = Node(3)
    second = Node(5)
    second.next = Node(6)
    second.next.next = Node(4)
    result = addTwoLists(first, second)
    print("Sum List:", end=" ")
    printList(result)
